mbuild/compound.py

```diff
@@ -1041,6 +1041,66 @@ def center(self):
         if np.all(np.isfinite(self.xyz)):
             return np.mean(self.xyz, axis=0)
 
+    @property
+    def center_of_mass(self):
+        """Returns the center of mass of a Compound in cartesian coordinates.
+
+        Returns
+        -------
+        np.ndarray, shape=(3,), dtype=float
+           The cartesian center of the Compound based on its Particles and molecular weight.
+            """
+        guessed_elements = set()
+        mass_vector = np.array([0., 0., 0.])
+        total_mass = 0
+        for atom in self.particles():
+            name = ''.join(char for char in atom.name if not char.isdigit())
+            try:
+                AtomicNum[atom.name]
+            except KeyError:
+                element = element_by_name(atom.name)
+                if name not in guessed_elements:
+                    warn('Guessing that "{}" is element: "{}"'.format(atom, element))
+                    guessed_elements.add(name)
+            else:
+                element = atom.name
+            mass = Mass[element]
+            total_mass += mass
+            mass_vector += mass*atom.pos
+
+        return mass_vector/total_mass
+
+    @property
+    def _molecular_mass(self):
+        """The summation of the individual atomic weights within a Compound.
+
+        Returns
+        -------
+        molecular_mass : Float
+            The summation of the individual atomic weights within a Compound.
+
+        Example
+        -------
+        To return the molecular masses of the children/subcompounds of a Compound (i.e. benzene within a box
+        of benzene), this method can be combined with a looping procedure.
+
+        >> [benzene._molecular_mass for benzene in box_of_benzene.children]
+        [78.116, ... , 78.116]
+        """
+        return sum(map(lambda x: x._atomic_mass, self.particles()))
+
+    @property
+    def _atomic_mass(self):
+        try:
+            AtomicNum[self.name]
+        except KeyError:
+            element = element_by_name(self.name)
```

Comment:
Going along with the comment above, I'm not sure we should be guessing element names here. I'm imagining a scenario where a user is operating on a system of united-atom methanes and calls `system.molecular_mass` and the individual particles, which are CH4s, are interpreted as carbons. I think instead we should:

1. Have `_atomic_mass` first check to see if `self` is a `Particle` (i.e. if it has no children), otherwise return None

2. If `AtomicNum[self.name]` returns a `KeyError` have `_atomic_mass` return None

_Then_ `molecular_mass` can return None (along with a warning message) _if_ `_atomic_mass` has returned None for any of the particles.